plugin/evm/vm.go

```diff
@@ -1423,11 +1424,23 @@ func (vm *VM) conflicts(inputs set.Set[ids.ID], ancestor *Block) error {
 // getAtomicTx returns the requested transaction, status, and height.
 // If the status is Unknown, then the returned transaction will be nil.
 func (vm *VM) getAtomicTx(txID ids.ID) (*Tx, Status, uint64, error) {
+	// Check accepted blocks for atomic txs
 	if tx, height, err := vm.atomicTxRepository.GetByTxID(txID); err == nil {
 		return tx, Accepted, height, nil
 	} else if err != database.ErrNotFound {
 		return nil, Unknown, 0, err
 	}
+
+	// Check processing blocks for atomic txs
+	// We should do this check here because blocks can pick up these atomic txs
+	// prior to block acceptance and should have status as processing.
+	// These txs will be available in the mempool as well but could give misleading
+	// results if dropped by another processing block.
```

Comment:
```suggestion
	// We check the atomic backend here because processing blocks can
	// contain atomic txs and their status report as processing.
	// This takes precedence over the tx's mempool status, where
	// the status may be dropped (if it was included in a processing block).
```

Reply:
thanks

Reply:
thanks